### PR TITLE
Fixed #28575 -- Made RelatedObjectDoesNotExist classes pickable.

### PR DESCRIPTION
**Pull request by charettes**
This ended up easier than I thought @timgraham, sorry for stealing the remaining work from you @rmtobin!

### CODE REVIEW

**Comment by rmtobin** [comment]
No worries, I’m glad you were able to figure out a solution!

**Comment by timgraham** [comment]
I don't easily understand what you wrote in the commit message about the reason for the `__reduce__()` methods. Anyway, I think that explanation would be better as code comments.

**Comment by charettes** [comment]
@timgraham added docstrings to the `__reduce__` methods that match `Field.__reduce__`.

File: tests/queryset_pickle/tests.py [hunk 1]
```diff
@@ -55,6 +55,18 @@ def test_multipleobjectsreturned_class(self):
         klass = Event.MultipleObjectsReturned
         self.assertIs(pickle.loads(pickle.dumps(klass)), klass)
 
+    def test_forward_relatedobjectcoesnotexist_class(self):
```
Comment (jdufresne) [inline tests/queryset_pickle/tests.py]: Typo in both test names:

`relatedobjectcoesnotexist` -> `relatedobjectdoesnotexist`

Comment (charettes) [reply tests/queryset_pickle/tests.py]: Thanks, just fixed it.

**Pull request merged**